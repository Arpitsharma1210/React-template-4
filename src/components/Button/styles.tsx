import { styled } from "styled-components";
import {
  fontWeight,
} from "../../theme/style.typography";


export const StyledButtonTextWrapper = styled.div<{ $bold?: string }>`
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 10px;
  font-weight: ${({ $bold }) =>
    $bold === "true" ? "bold" : fontWeight.regular};
`;
