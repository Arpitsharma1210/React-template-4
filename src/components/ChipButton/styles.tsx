import { styled, css } from "styled-components";
import {
  fontSize,
  baseFontFamily,
  fontWeight,
} from "../../theme/style.typography.js";
import { brand, colors } from "../../theme/style.palette.js";

type ButtonEnum = {
  [key: string]: {
    color: string;
    bgColor: string;
    borderColor: string;
  };
};

const Buttons: ButtonEnum = {
  yellow: {
    color: colors.yellow10,
    bgColor: brand.primaryMain,
    borderColor: brand.secondaryMain,
  },
  yellow2: {
    color: colors.yellow10,
    bgColor: colors.yellow50,
    borderColor: colors.yellow10,
  },
  green: {
    color: colors.green5,
    bgColor: colors.green25,
    borderColor: colors.green5,
  },
  blue: {
    color: colors.blue10,
    bgColor: colors.blue25,
    borderColor: colors.blue10,
  },
  red: {
    color: colors.red10,
    bgColor: colors.red50,
    borderColor: colors.red10,
  },
  orange: {
    color: colors.orange10,
    bgColor: colors.orange25,
    borderColor: colors.orange10,
  },
  white: {
    color: colors.grey100,
    bgColor: colors.white25,
    borderColor: colors.grey100,
  },
  purple: {
    color: colors.purple10,
    bgColor: colors.purple25,
    borderColor: colors.purple10,
  },
  grey:{
    color: colors.blackExtra,
    bgColor: colors.grey10,
    borderColor: colors.grey100,
  }
};

export const StyledButton = styled.button<{ color: string; size?: string }>`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  gap: 10px;

  cursor: pointer;
  border-radius: 4px;
  font-weight: ${fontWeight.regular};
  font-family: ${baseFontFamily};
  ${({ color }) => css`
    border: 1px solid ${Buttons[color].borderColor};
    color: ${Buttons[color].color};
    background-color: ${Buttons[color].bgColor};
  `}
  ${({ size }) =>
    (size === "sm" &&
      css`
        min-height: 23px;
        padding: 2px 6px;
        font-size: ${fontSize.b3};
        line-height: 16px;
      `) ||
    (size === "md" &&
      css`
        min-height: 28px;
        padding: 4px 12px;
        font-size: ${fontSize.b2};
        line-height: 19px;
      `)}
`;
