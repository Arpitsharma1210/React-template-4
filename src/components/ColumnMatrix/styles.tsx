import styled from "styled-components";
import { colors } from "../../theme/style.palette";
import { baseFontFamily, fontSize } from "../../theme/style.typography";

export const ColumnsContainer = styled.div`
  display: flex;
  column-gap: 175px;
`;

export const Column = styled.div`
  display: flex;
  flex-direction: column;
  gap: 35px;
`;

export const IndividualField = styled.div`
  display: flex;
  flex-direction: column;
  gap: 4px;
`;

export const FieldName = styled.div`
  font-size: ${fontSize.b1};
  font-family: ${baseFontFamily};
  line-height: 28px;
`;

export const FieldValue = styled.div`
  color: ${colors.grey100};
  font-family: ${baseFontFamily};
  font-size: ${fontSize.b2};
  line-height: 20px;
`;
