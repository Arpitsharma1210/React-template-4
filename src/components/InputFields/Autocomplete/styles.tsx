import { styled, css } from "styled-components";
import { Autocomplete } from "@mui/material";
import { colors } from "../../../theme/style.palette";

export const StyledAutoComplete = styled(Autocomplete)<{
  formatfor: "table" | "form";
}>`
  ${({ formatfor }) =>
    (formatfor === "table" &&
      css`
        width: 280px;
        background: ${colors.white25};
        border-radius: 10px !important;

        & .MuiInputBase-root {
          padding: 0px 38px 0px 9px !important;
        }

        & .MuiOutlinedInput-notchedOutline {
          border: 1px solid ${colors.grey25};
          border-radius: 10px !important;
        }
      `) ||
    (formatfor === "form" &&
      css`
        width: 100%;
        min-width: 368px !important;
        & .MuiInputBase-input {
          padding: 14px 19px;
          height: 30px;
        }
        & .MuiOutlinedInput-notchedOutline {
          border: 1px solid ${colors.grey100};
          border-radius: 6px !important;
        }
      `)}
`;
