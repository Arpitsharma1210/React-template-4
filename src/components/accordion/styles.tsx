import { styled, css } from "styled-components";
import {
  Typography,
} from "@mui/material";
import { AddCircle, RemoveCircle } from "@mui/icons-material";
import { colors } from "../../theme/style.palette";

const IconCss = () => css`
  color: ${colors.black};
  font-size: 26px;
`;

export const AddIcon = styled(AddCircle)`
  ${IconCss()}
`;

export const CloseIcon = styled(RemoveCircle)`
  ${IconCss()}
`;

export const StyledTypography = styled(Typography)<{ expanded: boolean }>`
  &.MuiTypography-root {
    order: ${({ expanded }) => (expanded ? -69 : 69)};
  }
`;
