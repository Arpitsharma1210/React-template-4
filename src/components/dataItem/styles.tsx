import { styled } from "styled-components";
import { colors } from "../../theme/style.palette";
import {
  baseFontFamily,
  fontSize,
  fontWeight,
} from "../../theme/style.typography";

export const StyledDataItem = styled.div`
  display: flex;

  & .imgContainer {
    height: 72px;
    width: 72px;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    background: ${colors.yellowGradient};
    margin-right: 16.8px;

    & img {
      width: 30px;
      height: 30px;
      display: block;
    }
  }

  & .databox {
    & .title {
      color: ${colors.grey};
      font-family: ${baseFontFamily};
      font-size: ${fontSize.b3};
      font-weight: ${fontWeight.regular};
      line-height: 16.8px;
      margin-bottom: 4px;
    }

    & .data {
      color: ${colors.grey100};
      font-family: ${baseFontFamily};
      font-size: 28px;
      font-weight: ${fontWeight.semiBold};
      line-height: 28px;
      letter-spacing: -0.28px;
      margin-bottom: 4px;
    }

    & .stat {
      & span:first-child {
        font-size: 11px;
        font-weight: ${fontWeight.bold};
      }
      & span:nth-child(2) {
        font-size: 11px;
        font-weight: ${fontWeight.regular};
        line-height: ${fontSize.b2};
        margin-right: 1px;
      }
      & span:nth-child(3) {
        font-size: 12px;
        font-weight: ${fontWeight.regular};
        line-height: ${fontSize.b2}
      }
      &.inc {
        & span:first-child {
          color: ${colors.green5};
        }
      }
      &.dec {
        & span:first-child {
          color: ${colors.red10};
        }
      }
    }
  }
`;
