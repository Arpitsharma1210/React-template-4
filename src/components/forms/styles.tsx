import styled from "styled-components";
import { colors, brand } from "../../theme/style.palette";
import {
  fontSize,
  fontWeight,
  secondaryFontFamily,
} from "../../theme/style.typography";

export const StyledCandidateHeader = styled.h3`
  color: ${brand.secondaryMain};
  font-family: ${secondaryFontFamily};
  font-size: ${fontSize.h3};
  font-weight: ${fontWeight.medium};
  line-height: 50px;
  margin: 0 0 36px 0;
`;

export const UploadButton = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;

  position: relative;
  font-size: 16px;
  width: 100%;
  border-radius: 6px;
  border: 1px solid ${colors.yellow30};
  background: ${brand.primaryMain};

  & i {
    margin-right: 18px;
  }

  & input {
    opacity: 0;
    cursor: pointer;
    position: absolute;
    left: 0;
    top: 0;
    height: 100%;
    width: 100%;
  }
`;
