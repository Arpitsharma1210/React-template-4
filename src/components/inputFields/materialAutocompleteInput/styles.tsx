import { styled, css } from "styled-components";
import { Autocomplete } from "@mui/material";
import { brand, colors } from "../../../theme/style.palette";
import { fontSize, fontWeight } from "../../../theme/style.typography";

export const StyledAutoComplete = styled(Autocomplete)<{
  formatfor: "table" | "form";
}>`
  ${({ formatfor }) =>
    (formatfor === "table" &&
      css`
        width: 280px;
        background: ${colors.white25};
        border-radius: 10px !important;

        & .MuiInputBase-root {
          padding: 0px 38px 0px 9px !important;
        }

        & .MuiOutlinedInput-notchedOutline {
          border: 1px solid ${colors.grey25};
          border-radius: 10px !important;
        }
      `) ||
    (formatfor === "form" &&
      css`
        width: 100%;
        min-width: 368px !important;
        & .MuiInputBase-input {
          padding: 14px 19px;
          height: 30px;
        }
        & .MuiOutlinedInput-notchedOutline {
          border: 1px solid ${colors.grey100};
          border-radius: 6px !important;
        }
      `)}
`;
export const StyledChipContainer = styled.div`
    display: flex;
    padding: 2px 5px 2px 10px;
    justify-content: center;
    align-items: center;
    gap: 5px;
    border-radius: 60px;
    background: ${brand.primaryMain};
    margin-right : 5px;
`

export const StyledChipLabel = styled.span`
    color : ${brand.primaryMain};
    font-size : ${fontSize.b2};
    font-weight : ${fontWeight.medium};
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    max-width: 72px;

`

export const StyledChipCloseContainer = styled.div`
    cursor : pointer;
    display : flex;
`