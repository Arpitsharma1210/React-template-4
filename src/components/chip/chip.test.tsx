import React from 'react';
import { render, screen } from '@testing-library/react';
import Chip from './chip';
import ChipGroup from './chipGroup';
import renderer from 'react-test-renderer';
import { StyledPopoverChipGroup, StyledPopoverSectionContainer, StyledPopoverSectionHeader } from './styles';

describe('Chip Component', () => {
  it('renders chip with text correctly', () => {
    render(<Chip text="Test Chip" />);
    const chipElement = screen.getByText('Test Chip');
    expect(chipElement).toBeInTheDocument();
  });
});

describe('ChipGroup Component', () => {
  it('renders only specified number of chips when displayCount prop is provided', () => {
    const data = [{ header: 'Group  1', data: [{ key: '1', text: 'Chip  1', bgColor: 'red', textColor: 'white' }] }];
    const tree = renderer
      .create(<ChipGroup data={data} displayCount={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  
      it('renders only specified number of chips when displayCount prop is provided', () => {
        const data = [{ header: 'Group 1', data: [{ key: '1', text: 'Chip 1', bgColor: 'red', textColor: 'white' }] }];
        render(<ChipGroup data={data} displayCount={1} />);
        const chipElement = screen.getByText('Chip 1'); 
        expect(chipElement).toBeInTheDocument();
    });
    
    it('renders chip popover when more chips are available', () => {
        const data = [
          { header: 'Group 1', data: [{ key: '1', text: 'Chip 1', bgColor: 'red', textColor: 'white' }] },
          { header: 'Group 2', data: [{ key: '2', text: 'Chip 2', bgColor: 'blue', textColor: 'white' }] }
        ];
        render(<ChipGroup data={data} />);
        const popoverTrigger = screen.getByText(/(\+)(\d+)/); 
        expect(popoverTrigger).toBeInTheDocument();
    });    

    it('renders StyledPopoverChipGroup correctly', () => {
      const tree = renderer
        .create(<StyledPopoverChipGroup>Chip Group</StyledPopoverChipGroup>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders StyledPopoverSectionHeader correctly', () => {
      const tree = renderer
        .create(<StyledPopoverSectionHeader>Section Header</StyledPopoverSectionHeader>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders StyledPopoverSectionContainer correctly', () => {
      const tree = renderer
        .create(<StyledPopoverSectionContainer>Section Content</StyledPopoverSectionContainer>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders chip popover when more chips are available', () => {
      const data = [
        { header: 'Group  1', data: [{ key: '1', text: 'Chip  1', bgColor: 'red', textColor: 'white' }] },
        { header: 'Group  2', data: [{ key: '2', text: 'Chip  2', bgColor: 'blue', textColor: 'white' }] }
      ];
      const tree = renderer
        .create(<ChipGroup data={data} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders chip with icon when iconCtaClick prop is provided', () => {
      const tree = renderer
        .create(<Chip text="Test Chip" iconCtaClick={() => {}} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });