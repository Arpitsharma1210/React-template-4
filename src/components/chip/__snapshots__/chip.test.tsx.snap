// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipGroup Component renders StyledPopoverChipGroup correctly 1`] = `
<div
  className="sc-iCoHzw feAYRd"
>
  Chip Group
</div>
`;

exports[`ChipGroup Component renders StyledPopoverSectionContainer correctly 1`] = `
<div
  className="sc-jSFipO jwYrzW"
>
  Section Content
</div>
`;

exports[`ChipGroup Component renders StyledPopoverSectionHeader correctly 1`] = `
<span
  className="sc-gKAaef hNlCVQ"
>
  Section Header
</span>
`;

exports[`ChipGroup Component renders chip popover when more chips are available 1`] = `
<div
  className="MuiGrid-root css-popmxn-MuiGrid-root"
>
  <div
    className="sc-dlniIP kemozv"
    onClick={[Function]}
  >
    <div
      className="sc-bdnyFh hMiSEt"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 sc-gtsqUy jLghVx css-ahj2mt-MuiTypography-root"
      >
        +2
      </p>
    </div>
  </div>
</div>
`;

exports[`ChipGroup Component renders chip with icon when iconCtaClick prop is provided 1`] = `
<div
  className="sc-bdnyFh hMiSEt"
>
  <p
    className="MuiTypography-root MuiTypography-body1 sc-gtsqUy jLghVx css-ahj2mt-MuiTypography-root"
  >
    Test Chip
  </p>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
    data-testid="CloseRoundedIcon"
    focusable="false"
    onClick={[Function]}
    style={
      {
        "cursor": "pointer",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M18.3 5.71a.9959.9959 0 0 0-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
    />
  </svg>
</div>
`;

exports[`ChipGroup Component renders only specified number of chips when displayCount prop is provided 1`] = `
<div
  className="MuiGrid-root css-popmxn-MuiGrid-root"
>
  <div
    bgColor="red"
    className="sc-bdnyFh jIVdiD"
    textColor="white"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 sc-gtsqUy jLghVx css-ahj2mt-MuiTypography-root"
    >
      Chip  1
    </p>
  </div>
</div>
`;
